src/NHibernate/Persister/Collection/OneToManyPersister.cs

```diff
@@ -76,7 +76,7 @@ protected override SqlCommandInfo GenerateDeleteString()
 				update.SetJoin(ownerPersister.TableName, KeyColumnNames, KeyType, JoinColumnNames, ownerPersister.GetPropertyColumnNames(CollectionType.LHSPropertyName));
 			}
 
-			if (HasIndex)
+			if (HasIndex && !indexContainsFormula)
```

Comment:
This is from:
https://github.com/fredericDelaporte/nhibernate-core/blob/ee91396495a5e0874704f22c24ccde7412dafea0/src/NHibernate/Persister/Collection/OneToManyPersister.cs#L134

Reply:
Yeah sorry for letting my french spelling leaking.